### PR TITLE
Make EmptySet the instance rather than the class

### LINKED ISSUE #16337: EmptySet and UniversalSet without S
I had thought we cleaned up all the S objects, but I guess not. 

Right now we have

```py
>>> from sympy import *
>>> EmptySet
<class 'sympy.sets.sets.EmptySet'>
>>> UniversalSet
UniversalSet()
```

There are two issues:

- `EmptySet` should be the instance, not the class.

- UniversalSet should not print with `()`.

### PR DESCRIPTION
**Pull request by oscarbenjamin**
#### References to other Issues or PRs

Closes #16575
Fixes #16337

#### Brief description of what is fixed or changed

Makes EmptySet (as imported from sympy) be the singleton instance of the emptyset rather than the class. This brings EmptySet into line with other classes like Reals.

#### Release Notes

* sets
    * EmptySet (as imported from sympy) is now the singleton instance of the empty set rather than the class.

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
I've hit a snag here which is that this change causes the symengine tests to fail. I think that's because symengine is importing and using EmptySet as a class rather than an instance. This can be fixed in both sympy and smyengine but changing either independently will fail.

How can we achieve the change simultaneously in both?

**Comment by asmeurer** [comment]
CC @isuruf

File: sympy/utilities/tests/test_wester.py [hunk 1]
```diff
@@ -966,7 +966,7 @@ def test_M19():
 
 
 def test_M20():
-    assert solveset(sqrt(x**2 + 1) - x + 2, x) == EmptySet()
+    assert solveset(sqrt(x**2 + 1) - x + 2, x) == EmptySet.__class__()
```
Comment (asmeurer) [inline sympy/utilities/tests/test_wester.py]: Shouldn't this just be `EmptySet`?

**Comment by asmeurer** [comment]
Our end goal should be to do

```
diff --git a/sympy/core/tests/test_singleton.py b/sympy/core/tests/test_singleton.py
index 4eb24dae83..4bc47ceca7 100644
--- a/sympy/core/tests/test_singleton.py
+++ b/sympy/core/tests/test_singleton.py
@@ -84,4 +84,4 @@ def test_names_in_namespace():
             continue
 
         # Use is here because of complications with ==
-        assert any(getattr(S, name) is i or type(getattr(S, name)) is i for i in d.values()), name
+        assert any(getattr(S, name) is i for i in d.values()), name

```

Maybe we should fix this at the metaclass level somehow.

**Comment by isuruf** [comment]
I can make SymEngine work with both SymPy release and the new version depending on SymPy's version string.

**Comment by isuruf** [comment]
Let me know if this PR will be merged in SymPy 1.5 and if so I'll add the version checking for `>1.4` or not.

File: sympy/utilities/tests/test_wester.py (hunk 1 above)
Comment (oscarbenjamin) [reply sympy/utilities/tests/test_wester.py]: Yes...

**Comment by oscarbenjamin** [comment]
@isuruf the other problem is that I can't merge this in SymPy without disabling the symengine tests.

File: sympy/sets/sets.py [hunk 2]
```diff
@@ -466,15 +466,16 @@ def powerset(self):
 
         A power set of an empty set:
 
-        >>> A = EmptySet()
+        >>> from sympy import FiniteSet, EmptySet
+        >>> A = EmptySet
         >>> A.powerset()
         {EmptySet()}
```
Comment (isuruf) [inline sympy/sets/sets.py]: This printing is needs to be fixed, right?

File: sympy/printing/pretty/pretty_symbology.py [hunk 3]
```diff
@@ -509,6 +509,7 @@ def xsym(sym):
     'Intersection':            U('INTERSECTION'),
     'Ring':                    U('RING OPERATOR'),
     'Modifier Letter Low Ring':U('Modifier Letter Low Ring'),
+    'EmptySequence':           'EmptySequence',
```
Comment (oscarbenjamin) [inline sympy/printing/pretty/pretty_symbology.py]: Maybe there's a better unicode symbol for this

File: sympy/sets/sets.py (hunk 2 above)
Comment (oscarbenjamin) [reply sympy/sets/sets.py]: That's fixed now. Some other doctests needed changing.

**Comment by oscarbenjamin** [comment]
> Our end goal should be to do

That's done now. The only singletons that failed that test were EmptySet and EmptySequence and they are both changed here.

Having looked through this code I think it's very confusing having the same name for the class and the instance. It also creates import problems. Right now with this PR these give different objects:
```python
from sympy.sets import EmptySet      # instance
from sympy.sets.sets import EmptySet # class
```
I think it would be better to change the class to `EmptySetClass` and have `EmptySet` only refer to the instance. Then both can be importable from `sympy.sets.sets` and `sympy.sets.__init__.py` can just import EmptySet rather than this awkwardness:
https://github.com/sympy/sympy/blob/4b93506a88cb924785a0e1dc98bb15205d2d45b8/sympy/sets/__init__.py#L8-L15

**Comment by oscarbenjamin** [comment]
@isuruf there are new symengine failures that look like:
```
 sympy/physics/mechanics/tests/test_rigidbody.py:test_pendulum_angular_momentum 
Traceback (most recent call last):
  File "/home/travis/build/sympy/sympy/sympy/utilities/runtests.py", line 1315, in _timeout
    function()
  File "/home/travis/build/sympy/sympy/sympy/physics/mechanics/tests/test_rigidbody.py", line 100, in test_pendulum_angular_momentum
    I = inertia(R1, 0, m * a**2 / 3, m * a**2 / 3)
  File "/home/travis/build/sympy/sympy/sympy/physics/mechanics/functions.py", line 86, in inertia
    ol = sympify(ixx) * (frame.x | frame.x)
  File "symengine_wrapper.pyx", line 826, in symengine.lib.symengine_wrapper.Basic.__mul__
  File "symengine_wrapper.pyx", line 554, in symengine.lib.symengine_wrapper._sympify
  File "symengine_wrapper.pyx", line 436, in symengine.lib.symengine_wrapper.sympy2symengine
  File "/home/travis/build/sympy/sympy/sympy/physics/vector/dyadic.py", line 116, in __eq__
    other = _check_dyadic(other)
  File "/home/travis/build/sympy/sympy/sympy/physics/vector/dyadic.py", line 557, in _check_dyadic
    raise TypeError('A Dyadic must be supplied')
TypeError: A Dyadic must be supplied
```

**Comment by oscarbenjamin** [comment]
This PR is now finished but blocked by the symengine tests

**Comment by isuruf** [comment]
This is fixed in symengine master. I'll make a release soon.

File: sympy/solvers/solveset.py [hunk 4]
```diff
@@ -843,7 +843,7 @@ def solve_decomposition(f, symbol, domain):
             elif isinstance(y_s, Union):
                 iter_iset = y_s.args
 
-            elif isinstance(y_s, EmptySet):
+            elif isinstance(y_s, type(EmptySet)):
```
Comment (isuruf) [inline sympy/solvers/solveset.py]: Shouldn't this be `y_s is EmptySet`?

File: sympy/solvers/solveset.py [hunk 5]
```diff
@@ -3395,7 +3395,7 @@ def nonlinsolve(system, *symbols):
 
         # positive dimensional system
         res = _handle_positive_dimensional(polys, symbols, denominators)
-        if isinstance(res, EmptySet) and any(not p.domain.is_Exact for p in polys):
+        if isinstance(res, type(EmptySet)) and any(not p.domain.is_Exact for p in polys):
```
Comment (isuruf) [inline sympy/solvers/solveset.py]: Same here

File: sympy/solvers/solveset.py (hunk 4 above)
Comment (oscarbenjamin) [reply sympy/solvers/solveset.py]: Yes

File: sympy/solvers/solveset.py (hunk 5 above)
Comment (oscarbenjamin) [reply sympy/solvers/solveset.py]: Yes. I've checked that there are no other cases using `type(EmptySet)`.

**Comment by isuruf** [comment]
I'll have symengine conda packages ready tomorrow.

**Comment by oscarbenjamin** [comment]
Thanks very much @isuruf. This is now ready for review/merge.

**Comment by jksuom** [comment]
This looks good to me.

**Pull request merged**